drivers/can/Kconfig.mcp2515

```diff
@@ -34,7 +34,7 @@ config CAN_MCP2515_INT_THREAD_PRIO
 	  Priority level of the internal thread which is ran for
 	  interrupt handling and incoming packets.
 
-config CAN_MCP2515_MAX_FILTER
+config CAN_MAX_FILTER
```

Comment:
Oh right, the mcux and stm32 Kconfig have a ```depends on``` on the symbol. Thanks for sorting this out :-)

Reply:
The problem here is that the MCP2515 could be used as an additional CAN controller beside the internal. Then we would run into problems here!

Reply:
Correct me if I'm wrong but wouldn't using the MCP2515 as a 2nd CAN controller be exceptional?  I imagine the majority of users of the MCP2515 would be adding a CAN controller to a SOC that didn't have one.  With this change those users can build the CAN sample without a Kconfig warning regarding CAN_MAX_FILTER.

Reply:
True. Let's use CAN_MAX_FILTER for all drivers.

Reply:
Apologies I meant to send this earlier, I've been thinking about this and I've come around to agree with you that it would be nice to make it straightforward to use it as a 2nd CAN controller (I missed this intent when I first saw the Kconfig warning and was trying to understand the code).  It dawned on me yesterday there's another way to get rid of the compile warning with an extra conf file, it will require a bit more documentation, I'll try to push the changes tomorrow.

Reply:
The good thing about only using CAN_MAX_FILTER is that there is only once config for all boards and samples. It makes it easy to migrate apps.

Reply:
An alternate CAN sample conf file is now added for building the sample using the MCP2515 driver.